drift/lib/src/dsl/table.dart

```diff
@@ -62,6 +62,23 @@ abstract class Table extends HasResultSet {
   @visibleForOverriding
   Set<Column>? get primaryKey => null;
 
+  /// Override this to specify unique keys:
+  /// ```dart
+  /// class IngredientInRecipes extends Table {
+  ///  @override
+  ///  Set<Column> get uniqueKey => {recipe, ingredient};
+  ///
+  ///  IntColumn get recipe => integer()();
+  ///  IntColumn get ingredient => integer()();
+  ///
+  ///  IntColumn get amountInGrams => integer().named('amount')();
+  ///}
+  /// ```
+  /// The getter must return a set literal using the `=>` syntax so that the
+  /// drift generator can understand the code.
+  @visibleForOverriding
+  Set<Column>? get uniqueKey => null;
```

Comment:
I'm not sure if a single getter returning a set is the best way to expose this. After all, a table can have multiple different unique constraints.

The unique constraint on the column builder is a great addition for sure, but for this I think it would be better to have a generic `List<TableConstraint>` getter? Users could then do something like

```dart
List<TableConstraint> get tableConstraints => [
  unique(recipe, ingredient)
];
```

But of course this would be much harder to parse in the generator... Maybe there's a simpler solution that could still support multiple independent unique constraints?

Reply:
I think the best solution is a `List<Set<GeneratedColumn>> uniqueKeys => [{column1, column2}, {column1, column3}]` for multiple keys. `UNIQUE` is the most used table level constraint that's why I think it deserves DSL option. For FKs there is the column level `references()` already. I think we not need to cover all constraint option and add for ex. `ON CONFLICT` clause as DSL. There is the table and column level  `customConstraints` for special needs.

Reply:
Good point, I agree on the `List<Set<GeneratedColumn>>` type which should be able to express this well.

Reply:
`GeneratedColumn` is not works, because you have to self-reference columns that `Column`